### PR TITLE
VB-3296 Rewrite 'Visits by date' page to use new Orchestration service APIs

### PR DESCRIPTION
**Pull request by tpmcgowan**
This is a rewrite of the code behind the 'View visits by date' page. The template/markup of the page is largely unchanged. 

Previously, a lot of work was done in the front-end to get all visits for a day and work back to what the schedules were - as well as looking up prisoner and visitor details. This was complex and slow.

New endpoints on the Orchestration service mean that, for a given day, the scheduled session templates can be found and then visits looked up for these as they are selected. Adult/child counts are no longer needed so no lookups to Contact Registry. Pagination has been removed too, which removes a lot of code.

(As part of this PR there are also some fixes to make the 'Back' link work properly on the booking summary page. It is linked to from several places and was not consistently populating the Back link correctly.)

See also #742 (merged into this) for explanation of handling of 'Unknown' visits.

### CODE REVIEW

File: integration_tests/integration/visitsByDate.cy.ts [hunk 1]
```diff
@@ -14,22 +14,32 @@ context('View visits by date', () => {
...
-    TestData.contact(),
-    TestData.contact({
-      personId: 4322,
-      firstName: 'Bob',
-      dateOfBirth: childDob,
-      relationshipCode: 'SON',
-      relationshipDescription: 'Son',
+  const prisonId = 'HEI'
+
+  const sessionSchedule = [
```
Comment (tpmcgowan) [inline integration_tests/integration/visitsByDate.cy.ts:19]: This integration test looks to have a lot of changes but it's mainly new mock APIs - actual assertions the same with a bit of tidying

File: server/routes/search.ts [hunk 2]
```diff
@@ -144,7 +144,10 @@ export default function routes({ auditService, prisonerSearchService, visitServi
   })
 
   get('/visit/results', async (req, res) => {
-    const { searchBlock1, searchBlock2, searchBlock3, searchBlock4 } = req.query
+    const searchBlock1 = typeof req.query.searchBlock1 === 'string' ? req.query.searchBlock1 : ''
```
Comment (tpmcgowan) [inline server/routes/search.ts:147]: Ensure these actually are strings and keep TypeScript happy.

File: server/routes/visit.test.ts [hunk 3]
```diff
@@ -254,73 +254,82 @@ describe('/visit/:reference', () => {
         })
     })
 
-    it('should render full booking summary page with visit information and prisoner tab selected with search back link when from visits', () => {
-      const url =
-        '/visit/ab-cd-ef-gh?query=startDate%3D2022-05-24%26type%3DOPEN%26time%3D3pm%2Bto%2B3%253A59pm&from=visit-search'
+    describe('back links', () => {
```
Comment (tpmcgowan) [inline server/routes/visit.test.ts:257]: Tests added for all the possible 'Back' link permutations on the booking summary page.

File: server/views/pages/visit/summary.njk [hunk 4]
```diff
@@ -22,12 +22,16 @@
 {% set pageTitle = applicationName + " - " + pageHeaderTitle + " - " + visit.reference %}
 
 {% if prisoner %}
-  {% if fromPage == 'visit-search' %}
-    {% set backLinkHref = "/visits?" + fromVisitSearchQuery %}
+  {% if fromPage == 'upcoming-visits' %}
```
Comment (tpmcgowan) [inline server/views/pages/visit/summary.njk:25]: Adding various missing potential back link sources.

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**